src/deep-resource/test/Resource.js

```diff
@@ -4,180 +4,115 @@ import chai from 'chai';
-    } catch (e) {
-      error = e;
-    }
+
+    KernelFactory.create({
+      Cache: Cache,
+      Security: Security,
+      Resource: Resource,
+    }, callback);
   });
 
   test('Check constructor sets _resources', function() {
-    chai.expect(Object.keys(resource._resources)).to.be.eql(['hello.world.example', 'deep.ng.root']);
-    chai.expect(resource._resources).to.be.not.eql({});
+    chai.expect(Object.keys(resource._resources)).to.be.eql(
+      ['hello.world.example', 'deep.ng.root']
+    );
   });
 
-
   test('Check has() method returns false', function() {
-    let error = null;
-    let actualResult = null;
-    try {
-      resource.bind(microserviceInstance);
-      actualResult = resource.has('invalid_resource_identifier');
-    } catch (e) {
-      error = e;
-    }
-
-    chai.expect(error).to.be.equal(null);
-    chai.expect(actualResult).to.be.equal(false);
+    resource.bind(microserviceInstance);
+    chai.expect(resource.has('invalid_res_identifier')).to.be.equal(false);
   });
 
   test('Check has() method returns true', function() {
-    let error = null;
-    let actualResult = null;
-    try {
-      resource.bind(microserviceInstance);
-      actualResult = resource.has(resourceName);
-    } catch (e) {
-      error = e;
-    }
-
-    chai.expect(error).to.be.equal(null);
-    chai.expect(actualResult).to.be.equal(true);
+    resource.bind(microserviceInstance);
+    chai.expect(resource.has(resourceName)).to.be.equal(true);
   });
 
   test('Check get() method returns valid object', function() {
-    let error = null;
-    let actualResult = null;
-    try {
-      actualResult = resource.get(`@${microserviceIdentifier}:${resourceName}`);
-    } catch (e) {
-      error = e;
-    }
-
-    chai.expect(error).to.be.eql(null);
-    chai.expect(actualResult.name).to.be.eql(resourceName);
-    chai.expect(Object.keys(actualResult._rawActions)).to.be.
-      contains(Object.keys(testResources[microserviceIdentifier][resourceName])[0]);
+    let actualResult = resource.get(
+      `@${microserviceIdentifier}:${resourceName}`
+    );
+
+    chai.expect(actualResult.name).to.be.equal(resourceName);
+    chai.expect(actualResult._rawActions).to.be.eql(
+      backendConfig.microservices[microserviceIdentifier].resources[resourceName]
+    );
   });
 
-  test('Check get() method throws \'MissingResourceException\' exception', function() {
-    let error = null;
-    try {
-      resource.get(`@${microserviceIdentifier}:invalid_resource_name`);
-    } catch (e) {
-      error = e;
+  test('Check get() method throws \'MissingResourceException\' exception',
+    function() {
+      let error = null;
+      try {
+        resource.get(`@${microserviceIdentifier}:invalid_resource_name`);
+      } catch (e) {
+        error = e;
+      }
+
+      chai.expect(error).to.be.not.equal(null);
+      chai.assert.instanceOf(error, MissingResourceException,
+        'error is an instance of MissingResourceException');
```

Comment:
- there is no need to check expect(error).to.be.not.equal(null) is error is checked to be instanceOf(error, MissingResourceException